lib/utility_functions.rb

```diff
@@ -20,7 +20,11 @@ def range_from_timeframe(timeframe, from = nil, to = nil)
     when "year"
       0.year.ago.beginning_of_year..0.year.ago.end_of_year
     when "custom"
-      convert_to_date(from)..convert_to_date(to)
+      if from && to
+        convert_to_date(from)..convert_to_date(to)
+      else
+        0.year.ago.beginning_of_year..0.year.ago.end_of_year
+      end
```

Comment:
Use guard clause instead of conditional statements
```suggestion
      return  convert_to_date(from)..convert_to_date(to) if from && to
      
      0.year.ago.beginning_of_year..0.year.ago.end_of_year
```